pkg/metrics/metrics.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"flag"
+	"net/http"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/netd/pkg/metrics/collector"
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var registry *prometheus.Registry
+
+type metricsConfig struct {
+	filter        string
+	listenAddress string
+	procPath      string
+}
+
+var mcfg metricsConfig
+
+func init() {
+	flag.StringVar(&mcfg.filter, "metrics-collectors", "", "Enable given metrics collectors (options: conntrack,socket).")
+	flag.StringVar(&mcfg.listenAddress, "metrics-port", ":8000", "Port on which to expose metrics.")
+	flag.StringVar(&mcfg.procPath, "metrics-proc-path", "/proc", "Proc directory to read metrics.")
```

Comment:
each proc file read probably should be an individual flag.

Reply:
All stat files used right now has only one common proc directory, so one shall be good.